common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -4,54 +4,108 @@
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class GuildModel extends Model {
-    private static final Pattern GUILD_NAME_MATCHER = Pattern.compile("§3(.*?)§b.*");
-    private static final Pattern GUILD_RANK_MATCHER = Pattern.compile("§7Rank: §f(.*)");
+    // Test suite: https://regexr.com/7hob9
+    private static final Pattern GUILD_NAME_MATCHER = Pattern.compile("§3([a-zA-Z ]*?)§b \\[[a-zA-Z]{3,4}]");
 
-    private String name = "";
-    private GuildRank rank;
+    // Test suite: https://regexr.com/7hobf
+    // Recruiter is intentionally first, otherwise Recruit will incorrectly match first
+    // This applies to all other patterns in this class involving guild ranks
+    private static final Pattern GUILD_RANK_MATCHER =
+            Pattern.compile("§7Rank: §f(Recruiter|Recruit|Captain|Strategist|Chief|Owner)");
+
+    // Test suite: https://regexr.com/7hoae
+    private static final Pattern MSG_LEFT_GUILD = Pattern.compile("§3You have left §b[a-zA-Z ]*§3!");
+
+    // Test suite: https://regexr.com/7hoah
+    private static final Pattern MSG_JOINED_GUILD = Pattern.compile("§3You have joined §b([a-zA-Z ]*)§3!");
+
+    // Test suite: https://regexr.com/7hoak
+    private static final Pattern MSG_RANK_CHANGED = Pattern.compile(
+            "'s guild rank from (?:Recruiter|Recruit|Captain|Strategist|Chief|Owner) to (Recruiter|Recruit|Captain|Strategist|Chief|Owner)");
+
+    private String guildName = "";
+    private GuildRank guildRank;
 
     public GuildModel(CharacterModel characterModel) {
         super(List.of(characterModel));
     }
 
+    // This needs to run before any chat modifications (eg. chat mentions, filter, etc)
+    // Side note; it is currently impossible to detect when we get kicked as there are no messages sent at all
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChatMessage(ChatMessageReceivedEvent e) {
+        StyledText message = StyledText.fromComponent(e.getMessage());
+
+        if (message.matches(MSG_LEFT_GUILD)) {
+            guildName = "";
+            guildRank = null;
+            WynntilsMod.info("User left guild");
+            return;
+        }
+
+        Matcher joinedGuildMatcher = message.getMatcher(MSG_JOINED_GUILD);
+        if (joinedGuildMatcher.matches()) {
+            guildName = joinedGuildMatcher.group(1);
+            guildRank = GuildRank.RECRUIT;
+            WynntilsMod.info("User joined guild " + guildName + " as a " + guildRank);
+            return;
+        }
+
+        if (message.startsWith("§3[INFO]§b ") && message.contains(McUtils.playerName() + "'s guild rank from")) {
+            // Username is checked first with simple string comparison
+            // Both versions of the 's string are checked as they will not include the color code
+            // Then determine rank with regex
+            // Username cannot be included in regex as at compile time it is unknown
```

Comment:
No, but you could add a `(.+)` (or more specific `([a-zA-Z0-9_]+)` or whatever are valid user name characters) and then check the group that it is equal to the player name.

That would be preferable, this is hard to read and understand what we are matching.